TOKEN_GUIDE.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Build a Token Generator with Astro
+description: Learn how you can generate tokens using SSR frameworks like Astro.
+---
+
+I love building websites with Astro. It's my favorite web framework, and I don't want to use any other one. However, I must stress test it and ensure Astro is flexible enough to fully commit.
+
+Astro is created to be the perfect use case for a content-heavy website. The opposite of that is a backend, and in this guide, we will build a backend token generator to be used for Agora video calls.
+
+Spoilers. My love for Astro has grown deeper. 
+
+![Token Flow](/assets/token-flow.png)
+
+## Prerequisites
+1. NodeJS and Astro installed.
+2. A developer account with [Agora](https://console.agora.io/).
+
+## Project Setup
+To create a new Astro project, run `npm create astro@latest`. Follow all the default options during the setup process, except use an empty template. This way, our project will be free of distractions. Then, we must install the Agora Token package to generate our tokens. You can do that using `npm i agora-token`.
+
+You must also create a `.env` and add `APP_ID` and `APP_CERTIFICATE` values, which you can find within your Agora Console.
+
+## What is a Token?
+Tokens are a form of authentication that verifies whether the user can access parts of an application. In this case, we want to use them to authenticate the user and their privileges within a video call. 
+
+The token is generated based on the HMAC-SHA256 cryptographic algorithm. This algorithm is an industry standard for authentication. 
+
+We don't need to dive into cybersecurity theory; just know that it's a good algorithm that will give you a safe token, and we can generate this token using the `agora-token` package.
+
+## Setting up an Endpoint in Astro
+To create an endpoint in Astro, you need to add a Javascript or Typescript file to the `pages` directory. This file should have the file name, followed by the return datatype extension, followed by `.js` or `.ts`.The Javascript and Typescript extensions will be removed during the build process.
+
+The most common type of return data is JSON, so our file will be a `.json.ts` file. We also need to pass some parameters. In Astro, you do this using brackets for the file/folder name. 
+
+Our file path for generating tokens will be `pages/rtc/[channel]/[role]/[uid].json.ts`, which will result in the end point of `rtc/{channel}/{role}/{uid}.json`.
+
+## Define the endpoint
+Our endpoint will be a `GET` request because our objective is to retrieve a token. You can define this in Astro by exporting a `GET` function and returning a JSON Response.
+
+We also import the `agora-token` package and bring in our `APP_ID` and `APP_CERTIFICATE`, which we defined in the `.env` file.
```

Comment:
Why do we need AgoraToken here? It feels like you can skip this here and add it to the next section where you use it to create the token.

Reply:
I thought it might be easier to bring in all the dependencies here. I'm not opposed to moving the `agora-token` to later, but then along the same lines I should probably move the APP_ID and certificate as well. 

I feel like it's a tradeoff because then the later code blocks get more complicated. Before I was showing only the logic for the specific part, but now with imports, would need to be indented, and showing the GET definition line, and then maybe the response even?

I'm not confident in either solution, but the tradeoff seemed to favor keeping it as it is.

Reply:
Fair points, let's leave it for now and discuss in our next sync.